src/content/docs/dropins/product-details/pdp-initialization.mdx

```diff
@@ -0,0 +1,96 @@
+---
+title: PDP dropin initialization
+description: Learn about the configuration options for the PDP initializer.
+---
+
+import Diagram from '@components/Diagram.astro';
+import Callouts from '@components/Callouts.astro';
+import OptionsTable from '@components/OptionsTable.astro';
+import Aside from '@components/Aside.astro';
+import { Steps } from '@astrojs/starlight/components';
+
+Customizing the PDP dropin initializer can help you meet your project requirements and use cases.
+
+## Configuration options
+
+The PDP dropin initializer allows you to define the language definitions, default locale, and models that your project will use.
+
+<OptionsTable
+  options={[
+    ['Option', 'Type', 'Req?', 'Description'],
+    ['langDefinitions', 'object', 'No', 'Provides language definitions for internationalization (i18n).'],
+    ['defaultLocale', 'string', 'No', 'Specifies the default locale. If not provided, defaults to en-US.'],
+    ['models', 'object', 'No', 'Defines the data structure and initial data. Options include initialData, transform, and fallBackData.'],
+  ]}
+/>
+
+:::note
+Models are used to define the data structure and initial data for the PDP dropin. The `initialData` object contains product details, options, and attributes.
+:::
+
+## Example
+
+The following code shows the default implementation of the PDP dropin initializer in this Commerce boilerplate:
+
+```typescript
+// Define the models object with a ProductDetails property
+const models = {
+ProductDetails: {
+  initialData: { ...product },
+  },
+};
+
+// Initialize Dropins
+initializers.register(productApi.initialize, {
+  langDefinitions,
+  models,
+});
+```
+
+## Set default product options
+
+When a user navigates to the product detail page (PDP) on your site, you can set certain options as preselected defaults. This use case allows merchandisers to set default options through the Adobe Commerce Admin, which provides a more customized and streamlined shopping experience.
+
+* **Simple products**—Default option selection is supported out-of-the-box through the Admin (`isDefault`). No additional PDP dropin customization is required.
+
+* **Complex products**—Default option selection is _not_ supported out-of-the-box. Instead, you must use [product attributes](https://experienceleague.adobe.com/en/docs/commerce-admin/catalog/product-attributes/product-attributes) and customize the PDP dropin initializers to define which attribute and value is used by default. If no option is defined as default, the first listed option will be preselected by default.
```

Comment:
This part is not entirely true - "the first listed option will be preselected by default".
This depends on the type of swatch used. Take this as an example -- https://experience-qa.adobe.net/solutions/magento-datalake-storefront-pdp/assets/qa20241016085457/standalone/product.html?sku=VSW12 -- this product does not have any default values, so no color swatch and no size are automatically selected.

The case that you mention is only happening for the dropdown type swatch. But even then, you can opt out of the preselection in the dropdown by providing a placeholder (eg: "Choose an option").

Reply:
I'll attempt a rewrite and re-request review.